src/uwtools/config/formats/base.py

```diff
@@ -22,15 +22,17 @@ class Config(ABC, UserDict):
     several configuration-file formats.
     """
 
-    def __init__(self, config_file: OptionalPath = None) -> None:
+    def __init__(self, config_file: OptionalPath = None, empty: bool = False) -> None:
```

Comment:
Add a flag to permit creating a config with an initially empty `data` `dict`. This is helpful for API uses cases where we have an in-memory dict and need a `Config` object based on it to conform to an interface requiring a `Config` argument, to e.g. to validate the config represented by the `dict`. It will be useful in near-future work allowing creation of `Config` objects via the API, specifically for cases where the user wants to start from an in-memory `dict`.

Similar support has been added in the subclasses of `Config`, below, where needed.